testplugins/src/main/java/org/spongepowered/test/ColorTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.test;
+
+import com.flowpowered.math.vector.Vector3d;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.type.DyeColor;
+import org.spongepowered.api.data.type.HandTypes;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.event.Listener;
+import org.spongepowered.api.event.filter.cause.First;
+import org.spongepowered.api.event.item.inventory.InteractItemEvent;
+import org.spongepowered.api.plugin.Plugin;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.util.Color;
+
+@Plugin(id = "color_test", name = "Color Test", description = "Right click to get an items color. Left click to set the color to orange.")
+public class ColorTest {
+
+    @Listener
+    public void onPrimaryClick(InteractItemEvent.Primary event, @First Player player) {
+        player.getItemInHand(HandTypes.MAIN_HAND).ifPresent(item -> {
```

Comment:
If this is a test, please make it toggle able either by command or configuration option. By no means should this always emit some message randomly causing people to freak out about debug messages when working on implementation ;).

Reply:
Tell that to whoever set up the inventory tests, I still get "You clicked on slot 7" messages every time I go to pull out an item xD. I'll switch those over to commands that can be used.

I'm not going to respond to the remaining reviews individually, but I'll address them in my edits tonight. I think I had a bit to much fun with rainbows (though they are fun) and let some of the good practices slip away from me. @ryantheleach, could you supply some info on what you'd like to see as a standard for test plugins so I can do that now and not have to rebuild it later?

Reply:
You won't have to rebuild it later, just make it somewhat toggleable to pass review, this is something larger that I havn't fully fleshed out yet.